### PR TITLE
Set optimizer number of crowds automatically

### PR DESCRIPTION
**Pull request by markdewing**
If "opt_num_crowds" is not set in the input file, set it to the number of threads.   The is the same as how batched VMC sets the number of crowds.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

laptop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCDrivers/WFOpt/QMCFixedSampleLinearOptimizeBatched.cpp [hunk 1]
```diff
@@ -613,6 +614,11 @@ bool QMCFixedSampleLinearOptimizeBatched::processOptXML(xmlNodePtr opt_xml,
 
   vmcEngine->enable_sample_collection();
 
+  if (opt_num_crowds_ == 0)
```
Comment (ye-luo) [inline src/QMCDrivers/WFOpt/QMCFixedSampleLinearOptimizeBatched.cpp:617]: Please add a call to checkNumCrowdsLTNumThreads()

File: src/QMCDrivers/WFOpt/QMCOptimizeBatched.cpp [hunk 2]
```diff
@@ -149,12 +150,19 @@ void QMCOptimizeBatched::process(xmlNodePtr q)
...
-  param_set.add(num_opt_crowds, "opt_num_crowds", "int");
+  param_set.add(opt_num_crowds, "opt_num_crowds", "int");
   param_set.put(q);
 
+  if (opt_num_crowds == 0)
+    opt_num_crowds = Concurrency::maxCapacity<>();
+
+  app_log() << " Number of crowds for optimizer: " << opt_num_crowds << std::endl;
+
+
```
Comment (ye-luo) [inline src/QMCDrivers/WFOpt/QMCOptimizeBatched.cpp:164]: Please reduce empty lines.

**Review by ye-luo — DISMISSED** [review]

**Comment by ye-luo** [comment]
@markdewing I halted the over crowded CI. Meanwhile please handle the merge conflict.

**Review by PDoakORNL — APPROVED** [review]
LGTM

**Pull request merged**